### ISSUE TITLE
Support for different output types for a model

### ISSUE THREAD

**Issue by Plyb** [DRAGNLabs/MIRROR-Pipeline#60]
Right now all MirrorModels are configured to return just a loss. That's fine for training, but for evaluation purposes, we'll need the actual output of the model.

**Acceptance Criteria**:
- MirrorModels can output different kinds of output (text logits, emotion classification, etc)
- The loss is still in a consistent place so it can be found by the trainer
- If at all possible, type safety is preserved (probably using generics)
